oasis-reference/src/main/kotlin/com/empowerops/volition/ref_oasis/OptimizerController.kt

```diff
@@ -1,210 +1,293 @@
+    @FXML internal lateinit var paramTreeView: TreeTableView<Parameter>
+    @FXML internal lateinit var nameColumn: TreeTableColumn<Parameter, String>
+    @FXML internal lateinit var valueColumn: TreeTableColumn<Parameter, String>
+    @FXML internal lateinit var lbColumn: TreeTableColumn<Parameter, Double>
+    @FXML internal lateinit var upColumn: TreeTableColumn<Parameter, Double>
+    @FXML lateinit var optimizerStatusLabel: Label
+    @FXML lateinit var descriptionLabel: Label
+    @FXML lateinit var selectedNodeInfoBox: VBox
+    @FXML lateinit var timeOutTextField: TextField
+    @FXML lateinit var useTimeout: CheckBox
+    @FXML lateinit var connectionListContainer: VBox
+    @FXML lateinit var issuesTextArea: TextArea
+
+    private val list: ObservableList<String> = FXCollections.observableArrayList()
+    private val messageList: ObservableList<Message> = FXCollections.observableArrayList()
+    private val resultList: ObservableList<Result> = FXCollections.observableArrayList()
+    private val currentEvaluationStatus = SimpleStringProperty()
+    private val issues = SimpleStringProperty()
+
+    private lateinit var connectionsView: ConnectionsView
+    private lateinit var control: OptimizerEndpoint
+    private lateinit var modelService: DataModelService
+    private lateinit var inputRoot: TreeItem<Parameter>
+    private lateinit var outputRoot: TreeItem<Parameter>
 
     enum class Type {
         Input, Output, Root
     }
 
-    data class Parameter(
-            val name: String,
-            val type: Type,
-            val value : Double? = null,
-            val lowerBound: Double? = null,
-            val upperBound: Double? = null
-    )
-
-    fun buildTree(config: Simulation): TreeItem<Parameter> {
+    private fun buildTree(config: Proxy?): TreeItem<Parameter> {
         val root = TreeItem<Parameter>(Parameter("root", Type.Root))
-        val root1 = TreeItem<Parameter>(Parameter("Inputs", Type.Root))
-        val root2 = TreeItem<Parameter>(Parameter("Outputs", Type.Root))
+        inputRoot = TreeItem(Parameter("Inputs", Type.Root))
+        outputRoot = TreeItem(Parameter("Outputs", Type.Root))
+        if(config==null) return root
 
         val inputs: List<TreeItem<Parameter>> = config.inputs.map {
-            TreeItem(Parameter(it.name, Type.Input, it.currentValue, it.lowerBound, it.upperBound))
+            val value = Parameter(it.name, Type.Input, it.currentValue)
+            value.lowerBound = it.lowerBound
+            value.upperBound = it.upperBound
+            TreeItem(value)
         }
         val outputs: List<TreeItem<Parameter>> = config.outputs.map {
             TreeItem(Parameter(it.name, Type.Input))
         }
 
-        root1.isExpanded = true
-        root2.isExpanded = true
+        inputRoot.isExpanded = true
+        outputRoot.isExpanded = true
 
-        root1.children.addAll(inputs)
-        root2.children.addAll(outputs)
+        inputRoot.children.addAll(inputs)
```

Comment:
Extracted as filed only because the commit logic, the other one can go back the way it was